### PR TITLE
Improvements to the blob cache manager

### PR DESCRIPTION
**Pull request by jiangliu**
Several improvements to the nydusd blob cache manager.

### CODE REVIEW

File: src/bin/nydusd/blob_cache.rs [hunk 1]
```diff
@@ -277,28 +336,76 @@ impl BlobCacheMgr {
...
+            );
+            let key = data_blob.get_key().to_string();
+
+            if let Err(e) = state.try_add(data_blob) {
+                // Rollback added bootstrap/data blobs.
+                let id = BlobCacheObjectId {
+                    domain_id: domain_id.to_string(),
+                    blob_id: id.to_string(),
+                };
+                let _ = state.remove(&id);
```
Comment (imeoer) [inline src/bin/nydusd/blob_cache.rs:386]: Do we need to rollback all added bootstrap/blob here?

File: src/bin/nydusd/blob_cache.rs [hunk 2]
```diff
@@ -277,28 +336,76 @@ impl BlobCacheMgr {
...
+                return Err(e);
+            }
+
+            // Associate the data blob with the bootstrap blob.
+            if let Some(BlobCacheObjectConfig::DataBlob(entry)) = state.id_to_config_map.get(&key) {
```
Comment (imeoer) [inline src/bin/nydusd/blob_cache.rs:391]: We can do `data_blob.clone()` before this to simplify the code.

**Comment by imeoer** [comment]
Others LGTM!

**Comment by hsiangkao** [comment]
@nan1994 please help test this with the latest fscache kernel, thanks!

File: src/bin/nydusd/blob_cache.rs (hunk 2 above)
Comment (jiangliu) [reply src/bin/nydusd/blob_cache.rs]: Thanks, fixed.

**Comment by nan1994** [comment]
I have tested this with the latest fscache kernel and it works well.

**Pull request merged**